scripts/extract_tybalt_weights.py

```diff
@@ -0,0 +1,182 @@
+
+# coding: utf-8
+
+# # Extracting Tybalt Weights
+# 
+# The weights learned by the Tybalt model indicate patterns of gene expression variably activated across tumors. As Tybalt is an unsupervised model, these weights learned can point to known biology, unknown biology, or unrelated noise. One of the benefits of a variational autoencoder (VAE), is that the weights learned that represent each encoding are  nonlinear. Therefore, they can extract out signal representative of noise while still retaining relevant known and unknown biology.
+# 
+# Here, we extract the weights learned by the VAE and save them to file. We also explore specific signals that should be present in the data:
+# 
+# 1. Signals representing the sex of the patient
+# 2. Melanoma activation patterns.
+
+# In[1]:
+
+import os
+import pandas as pd
+from keras.models import load_model
+
+import matplotlib.pyplot as plt
+import seaborn as sns
+
+
+# In[2]:
+
+sns.set(style='white', color_codes=True)
+sns.set_context('paper', rc={'font.size':8, 'axes.titlesize':10, 'axes.labelsize':15})   
+
+
+# In[3]:
+
+get_ipython().magic('matplotlib inline')
+plt.style.use('seaborn-notebook')
+
+
+# Because of the complex architecture involved in encoding the data, we will use the `decoded` weights to describe feature encoding specific activation patterns
+
+# In[4]:
+
+# Load the decoder model
+decoder_model_file = os.path.join('models', 'decoder_onehidden_vae.hdf5')
+decoder = load_model(decoder_model_file)
+
+
+# In[5]:
+
+# Load RNAseq file
+rnaseq_file = os.path.join('data', 'pancan_scaled_zeroone_rnaseq.tsv')
+rnaseq_df = pd.read_table(rnaseq_file, index_col=0)
+rnaseq_df.head(2)
+
+
+# In[6]:
+
+# For a future pathway analysis, the background genes are important
+# Also needed to set column names on weights
+background_file = os.path.join('data', 'background_genes.txt')
+background_genes = pd.DataFrame(rnaseq_df.columns)
+background_genes.to_csv(background_file, index=False, header=False, sep='\t')
+
+
+# ## Extract Tybalt weight matrix and write to file
+
+# In[7]:
+
+# Extract the weights from the decoder model
+weights = []
+for layer in decoder.layers:
+    weights.append(layer.get_weights())
+    
+weight_layer_df = pd.DataFrame(weights[1][0], columns=rnaseq_df.columns, index=range(1, 101))
+weight_layer_df.head(2)
+
+
+# In[8]:
+
+# Write the genes to file
+weight_file = os.path.join('results', 'tybalt_gene_weights.tsv')
+weight_layer_df.to_csv(weight_file, sep='\t')
```

Comment:
Do you want to save the indicies as well? if not add index=False here

Reply:
I do want the indices. But I agree that this should be specified to avoid indexing issues. I will set a name to the column indicating that it designates the encoding number.